src/liballoc/collections/btree/node.rs

```diff
@@ -627,10 +618,9 @@ impl<'a, K, V> NodeRef<marker::Mut<'a>, K, V, marker::Internal> {
     /// Adds a key/value pair and an edge to go to the right of that pair to
     /// the end of the node.
     pub fn push(&mut self, key: K, val: V, edge: Root<K, V>) {
-        // Necessary for correctness, but this is an internal module
-        debug_assert!(edge.height == self.height - 1);
-        debug_assert!(self.len() < CAPACITY);
-        debug_assert!(!self.is_shared_root());
+        assert!(edge.height == self.height - 1);
```

Comment:
Could be assert_eq! (but I guess that would generate extra dead code we can do without)